amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseTaker.java

```diff
@@ -536,40 +528,28 @@ private void updateAllLeases(Callable<Long> timeProvider)
     }
 
     /**
-     * Count leases by host. Always includes myself, but otherwise only includes hosts that are currently holding
-     * leases.
+     * Count leases by host, excluding all leases considered 'available to be taken' (ie unassigned or expired)
+     * using the supplied list. Always includes myself, but otherwise only includes hosts that are actively
+     * updating leases. There will be no entry for unassigned leases - ie the returned map will only have
+     * entries for non-null worker ids - so a good active-worker count is given by the size of the map.
      * 
-     * @param expiredLeases list of leases that are currently expired
+     * @param excludingAvailableLeases list of 'available' (unassigned/expired) leases to exclude from the count
      * @return map of workerIdentifier to lease count
      */
     @VisibleForTesting
-    Map<String, Integer> computeLeaseCounts(List<Lease> expiredLeases) {
-        Map<String, Integer> leaseCounts = new HashMap<>();
+    Map<String, Integer> computeActiveLeaseCountsByWorker(List<Lease> excludingAvailableLeases) {
         // The set will give much faster lookup than the original list, an
         // important optimization when the list is large
-        Set<Lease> expiredLeasesSet = new HashSet<>(expiredLeases);
+        Set<Lease> availableLeasesSet = new HashSet<>(excludingAvailableLeases);
 
-        // Compute the number of leases per worker by looking through allLeases and ignoring leases that have expired.
-        for (Lease lease : allLeases.values()) {
-            if (!expiredLeasesSet.contains(lease)) {
-                String leaseOwner = lease.leaseOwner();
-                Integer oldCount = leaseCounts.get(leaseOwner);
-                if (oldCount == null) {
-                    leaseCounts.put(leaseOwner, 1);
-                } else {
-                    leaseCounts.put(leaseOwner, oldCount + 1);
-                }
-            }
-        }
+        Map<String, Integer> activeLeaseCountsByWorker = allLeases.values().stream()
+            .filter(lease -> !lease.isUnassigned() && !availableLeasesSet.contains(lease))
+            .collect(groupingBy(Lease::leaseOwner, summingInt(lease -> 1)));
```

Comment:
Here filtering for `!lease.isUnassigned()` is what now stops the groupingBy `Lease::leaseOwner` from failing with the `NullPointerException: element cannot be mapped to a null key` error reported in https://github.com/awslabs/amazon-kinesis-client/pull/861 - given that unassigned leases are filtered out, there is no attempt to insert a `null` key entry in the map, so the code doesn't crash.

Reply:
Do we still want to keep this change given that #861 has been merged?